synapse/rest/client/account.py

```diff
@@ -878,9 +878,13 @@ def __init__(self, hs: "HomeServer"):
         self.auth = hs.get_auth()
 
     async def on_GET(self, request: SynapseRequest) -> Tuple[int, JsonDict]:
-        requester = await self.auth.get_user_by_req(request)
+        requester = await self.auth.get_user_by_req(request, allow_guest=True)
 
-        response = {"user_id": requester.user.to_string()}
+        response = {
+            "user_id": requester.user.to_string(),
+            # MSC: https://github.com/matrix-org/matrix-doc/pull/3069
+            "org.matrix.msc3069.is_guest": bool(requester.is_guest),
```

Comment:
For future reference: I'd have preferred this to be behind an opt-in config flag.

(see https://matrix-org.github.io/synapse/latest/development/experimental_features.html)

Reply:
It's not really a feature, but more of a very tedious spec bug fix.

Reply:
nevertheless, I'd prefer Synapse not to expose the unstable-named flag without people explicitly opting into it.